misc_docs/syntax/decorator_module_deprecated.mdx

```diff
@@ -0,0 +1,30 @@
+---
+id: "module-deprecated-decorator"
+keywords: ["deprecated", "decorator"]
+name: "@@deprecated"
+summary: "This is the `module deprecation` decorator."
+category: "decorators"
+---
+
+The `@@deprecated` decorator is used to indicate that a whole module is deprecated. This may be used by ReScript tooling to highlight deprecated code during development.
```

Comment:
```suggestion
The `@@deprecated` decorator is used to indicate that a whole module is deprecated. This may be used by ReScript tooling to highlight deprecated code during development.

The `@@deprecated` decorator is used to add a deprecation notes to the file-level of a module. The compiler and editor tooling will yield a warning whenever a deprecated file module is being used. This is useful to gracefully deprecate file modules and give your users a hint for future migration paths.
```